src/effects/VST3/VST3Utils.h

```diff
@@ -68,3 +77,70 @@ class VST3Utils final
    //Returns true on success.
    static bool ParseAutomationParameterKey(const wxString& key, Steinberg::Vst::ParamID& paramId);
 };
+
+
+struct VST3EffectSettings
+{
+   std::unordered_map<Steinberg::Vst::ParamID, Steinberg::Vst::ParamValue> mValues;
+};
+
+struct VST3Wrapper
+{
+   // For the time being, here we have only members that are needed
+   // to iterate parameters and extract preset state
+   //
+   Steinberg::IPtr<Steinberg::Vst::IEditController> mEditController;
+   Steinberg::IPtr<Steinberg::Vst::IComponent>      mEffectComponent;
+
+   using ParameterInfo = Steinberg::Vst::ParameterInfo;
+
+   using ParameterVisitor = std::function< bool(const ParameterInfo& pi) >;
+
+   mutable std::vector<ParameterInfo> mParameterInfos;
```

Comment:
Explain in a comment that this is mutable because it is cached once on demand.

Or even better:  define a class ParameterCache.  Give the wrapper a `const std::shared_ptr` to a cache.  Put the mutable member in the cache class, and give it a `const` member function that gives a reference to this vector.

Give `VSTWrapper` a constructor taking `std::shared_ptr` to this cache.  The constructor calls `make_shared` if the argument is null, or else uses the given argument.

These suggestions anticipate the definition of an Instance, which can share the cache of parameters with its plug-in.

Reply:
I tried what you suggest, but I came to a point where I had to use the mutable keyword anyway. 

The problem is, when it is time for constructing VST3Wrapper, mEditController is still null - if it was not, I could send it as an arg to the VST3Wrapper ctor, to be used to populate the cache - and there would no need to have a mutable something.

But the fact that the cache can be populated only when mEditController is not null, implies that the cache must be mutable.

Reply:
What I meant was, the member shared_ptr to the cache can be const, the cache object can be const, but then the members of the cache object need to be mutable.  There is still mutable, but hidden at a lower level.  Is that not what you had to do?

I recommend the shared_ptr so the plug-in and its instances can share just one cached definition of the set of parameters that is initialized only once, so we can be sure all visits are really over the same set of parameters.

I did something like this for AudioUnit, but the details did not require shared_ptr or mutable.